src/wwwroot/js/genpage/gentab/presets.js

```diff
@@ -292,16 +293,91 @@ function editPreset(preset) {
     fixPresetParamClickables();
 }
 
+let presetSortByElement;
+let presetSortReverseElement;
+let presetSortBy = localStorage.getItem('preset_list_sort_by') || 'Default';
+let presetSortReverse = localStorage.getItem('preset_list_sort_reverse') === 'true';
+
+
+function onPresetSortByChanged(ev) {
+    localStorage.setItem('preset_list_sort_by', presetSortBy = ev.target.value);
+    presetBrowser.update();
+}
+
+function onPresetSortReverseChanged(ev) {
+    localStorage.setItem('preset_list_sort_reverse', presetSortReverse = ev.target.checked);
+    presetBrowser.update();
+}
+
+function getSortInfo() {
+    const sortByElem = document.getElementById("preset_list_sort_by");
+    const reverseElem = document.getElementById("preset_list_sort_reverse");
+
+    if (sortByElem !== presetSortByElement) {
+        presetSortByElement?.removeEventListener("change", onPresetSortByChanged);
```

Comment:
whaaaat is going on here

Reply:
I was running into issues with the actual lifetime of the html elements vs when this code executes.  Finally wrote it to defensively check the element each time and add/remove listeners as needed.

Reply:
You can just not store it then, and use `getRequiredElementById`

Reply:
I again say go look at the image history code, which already handles this stuff.

If you make up a whole new way to solve a problem, that means if something bugs... there's now two different implementations that need to be separately looked into and separately fixed. If there's a feature to add, there's two different implementations that need to have that separately added to.

As much as possible, new things should either use, or exactly copy, the logic of pre-existing code. Not because one way is necessarily better than the other, but because it makes long term maintenance a lot easier.